bchrpc/bchrpc.proto

```diff
@@ -0,0 +1,401 @@
+syntax = "proto3";
+
+package pb;
+
+// bchrpc contains a set of RPCs that can be exposed publicly via
+// the command line options. This service could be authenticated or
+// unauthenticated.
+service bchrpc {
+
+    // Get info about the mempool.
+    rpc GetMempoolInfo(GetMempoolInfoRequest) returns (GetMempoolInfoResponse) {}
+
+    // GetBlockchainInfo info about the blockchain including the most recent
+    // block hash and height.
+    rpc GetBlockchainInfo(GetBlockchainInfoRequest) returns (GetBlockchainInfoResponse) {}
+
+    // Get info about the given block.
+    rpc GetBlockInfo(GetBlockInfoRequest)returns (GetBlockInfoResponse) {}
+
+    // Get a block.
+    rpc GetBlock(GetBlockRequest) returns (GetBlockResponse) {}
+
+    // Get a serialized block.
+    rpc GetRawBlock(GetRawBlockRequest) returns (GetRawBlockResponse) {}
+
+    // **Requires CfIndex**
```

Comment:
NIT: Prefer annotations like this to happen after the comment description.

```
// Get a block filter.
//
// ** Requires CfIndex**
```

Not a big deal if you prefer this syntax, but other parts of the codebase use the form above to describe locks and other notes.